### ISSUE TITLE
Reorganize SCSS sources

### ISSUE THREAD

**Issue by benjaoming** [django-wiki/django-wiki#1322]
It's hard to contribute to styling because the current layout of SCSS files doesn't reveal which files are original Bootstrap files (that should not be touched) and which are custom files (where changes should happen)

**Comment by oscarmcm** [comment]
Now, there's no real need to use a bunch of SCSS files, or are we going to continue supporting this?

https://getbootstrap.com/docs/4.6/getting-started/theming/

**Comment by benjaoming** [comment]
I would think that we still need to have the source Bootstrap files that we are building with? But we can split our own custom overrides from those so that everything from Bootstrap is kept separately and can easily be updated.

**Comment by benjaoming** [comment]
I think the layout that they suggest looks good:

```
your-project/
├── scss
│   └── custom.scss
└── bootstrap/
    ├── js
    └── scss
```

**Comment by oscarmcm** [comment]
> I would think that we still need to have the source Bootstrap files

Yes, we still need the imports like this:

```sass
// Custom.scss
// Option A: Include all of Bootstrap

// Include any default variable overrides here (though functions won't be available)

@import "../node_modules/bootstrap/scss/bootstrap";

// Then add additional custom code here
```

But I think that this will no longer be required to [have all of this](https://github.com/django-wiki/django-wiki/tree/main/src/wiki/static/wiki/bootstrap/scss) unless you still want to continue using the current structure.

**Comment by benjaoming** [comment]
I don't think we should have any Node stack in the project for _this_ reason. That'd have to happen because of something more important.

I have so far benefited from having the Bootstrap source files in the repo because I often read the code.

**Comment by benjaoming** [comment]
> But I think that this will no longer be required to [have all of this](https://github.com/django-wiki/django-wiki/tree/main/src/wiki/static/wiki/bootstrap/scss) unless you still want to continue using the current structure.

I want to continue having the Bootstrap source files but separate them from the files that contain overrides/additions.

I'd want to avoid any Node installation of Bootstrap for as long as possible, but certainly consider it if there were other things from Node :+1:

**Comment by benjaoming** [comment]
I want to also remove the CSS artifacts from the repo. I find that hatch has so-called "hooks" to ensure that something happens during the build process. We can use those hooks to build the SCSS.

I know that we currently have the `assets` command that we can run, but I'd like somehow that the build and release process doesn't come with a risk of un-synced assets by baking the assets building into the wheel/sdist building.

1) By writing our own hatch hook: https://hatch.pypa.io/1.9/plugins/build-hook/reference/
2) By using a plugin for hatch: https://github.com/rmorshea/hatch-build-scripts

Do you have any experience with this @oscarmcm ? :heart:

**Comment by oscarmcm** [comment]
Hey, sure I can take of it, and I think it's a good idea to move that outside of the scrips section. I'll be attending PyTexas this week so I'll try to get some time to work on it first.

**Comment by benjaoming** [comment]
That sounds fantastic! Thanks :heart:  I'll reorganize the SCSS either before or after this change, but I think we can keep them separate since it's just a matter of updating the `sassc` command with the new paths.

**Comment by benjaoming** [comment]
I want to begin by relocating the CSS and SCSS files in a separate change, but then I feel like the road is paved to start the BS5 upgrade :+1:

**Comment by oscarmcm** [comment]
@benjaoming maybe we don't need a new hook for this, maybe its just add a new dependency and ofc the proper configurations.

https://github.com/rmorshea/hatch-build-scripts

wdyt?

**Comment by benjaoming** [comment]
@oscarmcm that's indeed the dependency that I mentioned in the question ;) I'm not good at assessing whether to have a plugin instead of writing 1 hook... dependencies have a price in maintenance, risk of abandonment and supply chain security.. but then on the other hand, developing and maintaining a hook probably has disadvantages too :)

**Comment by oscarmcm** [comment]
oh sorry about not catching it right, gonna check what's the most easy solution for this without going crazy.

**Comment by benjaoming** [comment]
yes, agreed, it seems like it's immediately going to be easy to spend a disproportional amount of time on a solution for a little problem... it was very "intuitive" with a Makefile, yet here there seems to be an immediate uphill struggle.

**Comment by oscarmcm** [comment]
@benjaoming I was thinking on moving the Bootstrap files to a submodule, that way we can lock in the BS version inside the Git history without having all the files tracked.

I'll probably add some hatch scripts to sync the submodule as well.

**Comment by benjaoming** [comment]
@oscarmcm from experience, I would say that submodules are not nice. I understand that vendoring all their source files looks ugly, but in reality, they aren't updated so often and don't seem to cause issues.

I don't mean to say that having some NPM infrastructure to deal with Bootstrap releases is nicer. But checking out Bootstrap's repo also comes with a cost.

But do you think it's worth the effort?

We could also add a script to hatch that downloads and unzips a specific version of bootstrap releases? :)

**Comment by oscarmcm** [comment]
After talking with @ajpinedam (who has experience with this kind of upgrades) I’ll wait for his feedback on what would be the best approach here, indeed I think continuing with the same layout will be the most easiest approach! But as I always say nothing is written on stone and only time has the right answer… this might be the most clear way to do it.

**Comment by benjaoming** [comment]
Thanks for putting efforts into it! It's a very interesting challenge to solve, and something that I think more projects can benefit from. Looking forwards to getting cleverererer :)
